xml/libvirt_migrating_vms.xml

```diff
@@ -421,6 +421,13 @@ qemu+tls://&exampleuser_plain;@&wsIname;/system</screen>
 --migrateuri tcp://@&wsIname;:49152</screen>
         </listitem>
       </varlistentry>
+      <varlistentry>
+        <term>Offline migration transferring all used storage</term>
+        <listitem>
+<screen>&prompt.user;virsh migrate --offline --persistent --copy-storage-all \
```

Comment:
Dirty little secret: I've never tested this :-). Did you try it?

Reply:
No, If you have 2 hosts, can you try please? Otherwise I'll need to install a second VM host for that as it refuses to migrate on the same host even to a new user session. BTW it returns some strange error message for `--copy-storage-all` along the words: required operation is not valid - migration in deactivated state cannot manage non-shared storage

Reply:
Bummer, `--copy-storage-all` is not supported with offline migration:
# virsh migrate --offline --persistent --copy-storage-all tw-test-kvm qemu+ssh://virt81/system
error: Requested operation is not valid: offline migration cannot handle non-shared storage

Reply:
libvirt's offline migration support is limited. It appears to do the migration compatibility checks, but otherwise just copies the VM xml from source host to destination host. Storage would have to be copied separately.

Reply:
Thanks for testing Jim! Would live or non-live be an option?
 `virsh migrate --live --copy-storage-all`

Reply:
Yes, that works, although it requires use of libvirt storage pools :-(.
```
src # virsh migrate --live --persistent --copy-storage-all tw-test-kvm qemu+ssh://virt81/system
error: Storage pool not found: no storage pool with matching target path '/vm_images/jim/images/test-vm'
```
The VM on the source was created with virt-manager, so already had a storage pool for its single disk. I defined and started a similar pool on the destination host, and then was able to perform the migration. E.g.

```
dest # cat test-vm-pool.xml 
<pool type='dir'>
  <name>test-vm</name>
  <target>
    <path>/vm_images/jim/images/test-vm</path>
    <permissions>
      <mode>0755</mode>
      <owner>0</owner>
      <group>0</group>
    </permissions>
  </target>
</pool>
dest # virsh pool-define test-vm-pool.xml && virsh pool-start test-vm
```
The pool XML could be obtained from the source host with e.g. `src # virsh pool-dumpxml test-vm` and copied to the destination host.

Reply:
@jfehlig review my last update to this section please